backend/main.go

```diff
@@ -61,108 +66,87 @@ func (w *Worker) Run() {
 		w.log.Infof("Processing %d signature requests", len(signatureRequests))
 
 		for _, sr := range signatureRequests {
-			switch sr.IncomingType {
-			case store.Ethereum:
-				err := w.processIncomingEthereumSignatureRequest(sr)
-				if err != nil {
-					w.log.WithFields(log.F{"err": err, "hash": sr.IncomingTransactionHash}).
-						Error("Cannot process signature request")
+			var err error
+			switch sr.Action {
+			case store.Withdraw:
+				switch sr.DepositChain {
+				case store.Ethereum:
+					err = w.processStellarWithdrawalRequest(sr)
+				default:
+					err = fmt.Errorf("withdrawals for deposit chain %v is not supported", sr.DepositChain)
 				}
+			default:
+				err = fmt.Errorf("action %v is not supported", sr.Action)
+			}
 
-				w.log.WithField("hash", sr.IncomingTransactionHash).
-					WithField("network", sr.IncomingType).
-					Info("Processed signature request successfully")
-
-				err = w.Store.DeleteSignatureRequestForIncomingEthereumTransaction(context.TODO(), sr.IncomingTransactionHash)
-				if err != nil {
-					w.log.WithFields(log.F{"err": err, "hash": sr.IncomingTransactionHash}).
-						Error("Error removing signature request")
+			if err != nil {
+				w.log.WithFields(log.F{"err": err, "request": sr}).
+					Error("Cannot process signature request")
+				if p, ok := err.(problem.P); ok && p.Status == http.StatusBadRequest {
+					w.deleteRequest(sr)
 				}
+			} else {
+				w.log.WithField("request", sr).
+					Info("Processed signature request successfully")
+				w.deleteRequest(sr)
```

Comment:
I believe it is safe for a request to be processed multiple times because we still assert all the withdrawal preconditions when handling the signature request